### PR TITLE
Popupmenus

### PR DESCRIPTION
**Pull request by bercsenyirihi**
Trisztán also added a GetSpaceShips() to Asteroid, will need to be fixed.

### CODE REVIEW

File: java/src/hu/holyoil/view/popupmenus/TeleportGatePopupMenu.java [hunk 1]
```diff
@@ -1,16 +1,119 @@
...
+    public void rClick(Settler settler, TeleportGate teleporter){
+        JMenuItem travel = new JMenuItem("travel here");
+        travel.addActionListener(new moveListener(settler, teleporter));
+        this.add(travel);
+    }
+
+    /**
+     * Egy actionlistener, amely feladata hogy érzékelje amikor rákattintottak e majd végrehajtsa a mozgás műveletet.
+     */
+    public static class moveListener implements ActionListener {
```
Comment (vercibar) [inline java/src/hu/holyoil/view/popupmenus/TeleportGatePopupMenu.java:96]: no need to be static

File: java/src/hu/holyoil/view/popupmenus/SettlerActionPopupMenu.java [hunk 2]
```diff
@@ -1,16 +1,145 @@
...
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            settler.Mine();
+        }
+    }
+
+    /**
+     * Egy actionlistener, amely feladata hogy érzékelje amikor rákattintottak e majd végrehajtsa a fúrás műveletet.
+     */
+    public static class drillListener implements ActionListener {
```
Comment (vercibar) [inline java/src/hu/holyoil/view/popupmenus/SettlerActionPopupMenu.java:125]: no need to be static

File: java/src/hu/holyoil/view/popupmenus/SettlerActionPopupMenu.java [hunk 3]
```diff
@@ -1,16 +1,145 @@
...
+                mine.addActionListener(new mineListener(settler));
+                this.add(mine);
+            }
+        }
+    }
+
+    /**
+     * Egy actionlistener, amely feladata hogy érzékelje amikor rákattintottak e majd végrehajtsa a bányászás műveletet.
+     */
+    public static class mineListener implements ActionListener {
```
Comment (vercibar) [inline java/src/hu/holyoil/view/popupmenus/SettlerActionPopupMenu.java:101]: no need to be static

**Review by vercibar — CHANGES_REQUESTED** [review]
Two twings:

- Popup menus should hide themselves whenever an option gets clicked. Every listener's actionPerformed should contain a setVisible(false), otherwise they remain there after clicking them.
- The performed actions need to fire some kind of event that the gameframe can subscribe to. They need to signal to the environmentpanel for example that the map changed and that it needs to update its components.

**Comment by triszt4n** [comment]
Look at this beautiful child 🧒

**Review by vercibar — APPROVED** [review]

**Pull request merged**